fastly_compute/src/main.rs

```diff
@@ -181,13 +185,30 @@ async fn handle_request(worker: &SxgWorker, req: Request) -> Result<Response> {
 
 #[fastly::main]
 fn main(req: Request) -> Result<Response, std::convert::Infallible> {
+    let has_network_loop = req
+        .get_header_all_str("via")
+        .iter()
+        .any(|v| v.contains(VIA_SXGRS));
+    if has_network_loop {
+        return Ok(text_response("Network loop detected."));
+    }
     tokio::runtime::Builder::new_current_thread()
         .build()
         .unwrap()
         .block_on(async {
             let worker = create_worker().await;
-            let response = handle_request(&worker, req).await.unwrap_or_else(|msg| {
-                text_response(&format!("A message is gracefully thrown.\n{:?}", msg))
+            let response = handle_request(&worker, &req).await.unwrap_or_else(|_| {
+                let mut req = req;
+                match worker.get_fallback_url_and_cert_origin(req.get_url()) {
+                    Ok((fallback_url, _)) => {
+                        req.set_url(fallback_url);
+                    }
+                    Err(_) => {
+                        return text_response("Failed to construct fallback URL");
+                    }
+                };
+                req.append_header("via", VIA_SXGRS);
+                req.send(HTML_BACKEND_NAME).unwrap()
```

Comment:
I suppose there is a risk of panic from this `unwrap()`... is that OK? Does Fastly convert the error into a 500 response and keep serving subsequent requests with this app?

Alternatively, perhaps we could do `unwrap_or_else(|_| some_500_response("...")` here.

Reply:
Panicking leads to a 500 response. The service is still alive for subsequent requests.